File: java/exception-handler-example-api/src/main/java/com/example/infrastructure/notifications/NotificationContext.java
package com.example.infrastructure.notifications;

import java.util.Objects;

import com.example.infrastructure.notifications.Enums.NotificationType;

import java.util.ArrayList;

public class NotificationContext {

    public ArrayList<Notification> notifications;

    public NotificationContext() {
        this.notifications = new ArrayList<Notification>();
    }

    public ArrayList<Notification> getNotifications() {
        return notifications;
    }

    public boolean HasNotifications() {
        return Objects.nonNull(this.notifications) && this.notifications.size() > 0;
    }

    public void Add(NotificationType type, String message) {
        this.notifications.add(new Notification(type, message));
    }

    public void Add(Notification notification) {
        this.notifications.add(notification);
    }

    public void Add(ArrayList<Notification> notifications) {
        this.notifications.addAll(notifications);
    }

    public void AddNotFoundNotification(String message) {
        this.notifications.add(Notification.InexistentId(message));
    }

    public void AddValidationErrorNotification(String message) {
        this.notifications.add(Notification.ValidationError(message));
    }
}

File: CSharp/ExceptionHandlerExample.Api/Infraestructure/Notifications/NotificationContext.cs
using Infra.CrossCutting.Notifications.Enums;

namespace Infra.CrossCutting.Notifications
{
    public class NotificationContext 
    {
        public IList<Notification> notifications { get; set; }
        public bool HasNotifications => notifications.Any();

        public NotificationContext()
        {
            this.notifications = new List<Notification>();
        }

        public void Add(NotificationType type, string message)
        {
            this.notifications.Add(new Notification(type, message));
        }

        public void Add(Notification notification)
        {
            this.notifications.Add(notification);
        }

        public void Add(IList<Notification> notifications)
        {
            this.notifications.Concat(notifications);
        }

        public void AddNotFoundNotification(string message)
        {
            this.notifications.Add(new Notification().InexistentId<Notification>(message));
        }

        public void AddValidationErrorNotification(string message)
        {
            this.notifications.Add(new Notification().ValidationError<Notification>(message));
        }
    }
}